Update Consul ECS documentation with health sync changes (#11386)
hashicorp/consul: website/content/docs/ecs/architecture.mdx

The ACL controller stores all ACL tokens in AWS Secrets Manager, and tasks are configured to pull these
tokens from AWS Secrets Manager when they start.

### ECS Health Check Syncing

> Health checks are automatically synced from ECS into Consul.  This syncing
> automatically happens for all application containers that are marked as
> `essential`, have ECS `healthChecks`, and don't have native Consul health
> checks configured. The `mesh-init` container creates a TTL health check for
> every container that fits these criteria and the `health-sync` container ensures
> that the ECS and Consul health checks remain in sync.

Reviewer: ```suggestion
If the following conditions apply, ECS health checks automatically sync with Consul health checks for all application containers:

* marked as `essential` 
* have ECS `healthChecks` 
* are not configured with native Consul health checks 

The `mesh-init` container creates a TTL health check for
every container that fits these criteria and the `health-sync` container ensures
that the ECS and Consul health checks remain in sync.
```
Trying out a bulleted list for this information so that it's easier to scan - ignore if it doesn't seem like an improvement.
